### PR TITLE
update(auth): add username validation and uploading profile picture

### LINKED ISSUE #397: Update username validation rules in creating account
Apply the following rules in username

- Length between 4-32 characters.
- No space in the beginning and end(Trimed). Allow space in the middle of text.
- @ is not allowed.
- Only allow alphabetic, digits and punctuation(No emoji).

image

### PR DESCRIPTION
**Pull request by hyj1204**
### What this PR does 📖
- Add username validation in create account page
- Add uploading profile picture in create account page
- Update the username validation Regex(allow space in the middle of text)

https://user-images.githubusercontent.com/14248245/204443808-6a7303c7-9b11-47e3-8cf7-0d6d66047f1f.mov

### Which issue(s) this PR fixes 🔨
- Resolve #397

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

**Review by Jekrimo — APPROVED** [review]

File: src/components/prelude/auth/mod.rs [hunk 1]
```diff
@@ -65,18 +93,69 @@ pub fn Auth(cx: Scope<Props>) -> Element {
...
+                                .unwrap()
+                                .to_string();
+                                let parts_of_filename: Vec<&str> = filename.split('.').collect();
 
+                                let mime = match parts_of_filename.last() {
+                                    Some(m) => {
+                                        match *m {
+                                            "png" => IMAGE_PNG.to_string(),
+                                            "jpg" => IMAGE_JPEG.to_string(),
+                                            "jpeg" => IMAGE_JPEG.to_string(),
```
Comment (dariusc93) [inline src/components/prelude/auth/mod.rs:139]: You could do 

```rust
"jpg" | "jpeg" => IMAGE_JPEG.to_string()
```

Just to simplify it for that.

File: src/components/prelude/auth/mod.rs [hunk 2]
```diff
@@ -65,18 +93,69 @@ pub fn Auth(cx: Scope<Props>) -> Element {
...
+                                    Some(path) => path,
+                                    None => return
+                                };
+                                let file = match std::fs::read(&path) {
+                                    Ok(image_vec) => image_vec,
+                                    Err(_) => vec![],
+                                };
+                                let filename = std::path::Path::new(&path)
+                                .file_name()
+                                .unwrap_or_else(|| std::ffi::OsStr::new(""))
```
Comment (dariusc93) [inline src/components/prelude/auth/mod.rs:128]:
```suggestion
                                .map(|s| s.to_string_lossy().to_string())
                                .unwrap_or_default();
```

**Review by dariusc93 — APPROVED** [review]
Left some comments and a suggestion and will approve it as is since the logic can be done separately, but if the PR is suppose to support uploading a profile picture, I dont see the logic being used (though I could be overlooking it and it may be reusing the logic thats there). In this the case?

File: src/components/prelude/auth/mod.rs (hunk 2 above)
Comment (dariusc93) [reply src/components/prelude/auth/mod.rs:128]: Note this suggestion requires removing everything after the `upwrap_or_default()` so i would suggest applying this suggestion in the IDE.

**Review by InfamousVague — APPROVED** [review]

**Pull request merged**